src/tokens/ERC20RebaseDistributor.sol

```diff
@@ -260,16 +291,34 @@ abstract contract ERC20RebaseDistributor is ERC20 {
         // if `account` is rebasing, update its number of shares
         RebasingState memory _rebasingState = rebasingState[account];
         if (_rebasingState.isRebasing == 1) {
+            // compute rebased balance
             uint256 _rebasingSharePrice = rebasingSharePrice;
-            uint256 balanceAfter = (_rebasingState.nShares *
+            uint256 rebasedBalance = (_rebasingState.nShares *
                 _rebasingSharePrice) /
                 START_REBASING_SHARE_PRICE +
                 amount;
-            uint256 sharesAfter = (balanceAfter * START_REBASING_SHARE_PRICE) /
+            uint256 rawBalance = ERC20.balanceOf(account);
+            if (rebasedBalance < rawBalance) {
+                rebasedBalance = rawBalance;
+            }
+
+            // update number of shares
+            uint256 sharesAfter = (rebasedBalance * START_REBASING_SHARE_PRICE) /
                 _rebasingSharePrice;
             uint256 sharesReceived = sharesAfter - _rebasingState.nShares;
-            rebasingState[account].nShares = uint248(sharesAfter);
+            rebasingState[account] = RebasingState({
+                isRebasing: 1,
+                nShares: uint248(sharesAfter)
+            });
             totalRebasingShares = totalRebasingShares + sharesReceived;
+
+            // "realize" pending rebase rewards
+            uint256 mintAmount = rebasedBalance - rawBalance;
+            if (mintAmount != 0) {
+                ERC20._mint(account, mintAmount);
+                pendingRebaseRewards -= mintAmount;
```

Comment:
One thing I wonder is whether these `pendingRebaseRewards` decreases could ever underflow. I think _most likely not_ as `pendingRebaseRewards` is increased by the actual amount burned and then operations to determine the new share price round down, as well as the operations to "realize" rewards.  So, to be clear, I don't see an issue but will probably think more about it. IMO this doesn't block merging of this PR.